src/tiledbsoma_ml/dataset.py

```diff
@@ -203,155 +211,83 @@ def __init__(
+        if not obs_column_names:
+            raise ValueError("Must specify at least one value in `obs_column_names`")
 
-        # 3. Chunk and optionally shuffle chunks
         if self.shuffle:
-            assert self.io_batch_size % self.shuffle_chunk_size == 0
-            shuffle_split = np.array_split(
-                _gpu_split, max(1, ceil(min_len / self.shuffle_chunk_size))
-            )
-
-            # Deterministically create RNG - state must be same across all processes, ensuring
-            # that the joinid partitions are identical across all processes.
-            rng = np.random.default_rng(self.seed + self.epoch + 99)
-            rng.shuffle(shuffle_split)
-            obs_joinids_chunked = [
-                np.concatenate(b)
-                for b in batched(
-                    shuffle_split, self.io_batch_size // self.shuffle_chunk_size
+            # Verify `io_batch_size` is a multiple of `shuffle_chunk_size`
+            if self.io_batch_size % self.shuffle_chunk_size:
+                raise ValueError(
+                    f"{self.io_batch_size=} is not a multiple of {self.shuffle_chunk_size=}"
                 )
-            ]
-        else:
-            obs_joinids_chunked = np.array_split(
-                _gpu_split, max(1, ceil(len(_gpu_split) / self.io_batch_size))
-            )
 
-        # 4. Partition by DataLoader worker
-        n_workers, worker_id = get_worker_world_rank()
-        obs_splits = splits(len(obs_joinids_chunked), n_workers)
-        obs_partition_joinids = obs_joinids_chunked[
-            obs_splits[worker_id] : obs_splits[worker_id + 1]
-        ].copy()
-
-        if logger.isEnabledFor(logging.DEBUG):
-            partition_size = sum([len(chunk) for chunk in obs_partition_joinids])
-            logger.debug(
-                f"Process {os.getpid()} {rank=}, {world_size=}, {worker_id=}, n_workers={n_workers}, epoch={self.epoch}, {partition_size=}"
+        if self.seed is None:
+            object.__setattr__(
+                self, "seed", np.random.default_rng().integers(0, 2**32 - 1)
             )
 
-        return iter(obs_partition_joinids)
+        # Set distributed state
+        rank, world_size = get_distributed_world_rank()
+        object.__setattr__(self, "rank", rank)
+        object.__setattr__(self, "world_size", world_size)
 
-    def _init_once(self, exp: Experiment | None = None) -> None:
-        """One-time per worker initialization.
+    @property
+    def measurement_name(self) -> str:
+        return self.x_locator.measurement_name
 
-        All operations should be idempotent in order to support pipe reset().
+    @property
+    def layer_name(self) -> Optional[str]:
+        return self.x_locator.layer_name
 
-        Private method.
+    @property
+    def obs_joinids(self) -> NDArrayJoinId:
+        """Return this |ExperimentDataset|'s obs coordinates (possibly partitioned for a GPU process / |DataLoader|
```

Comment:
this docstring needs clarity on what it returns and when.  "possibly..." is not all that helpful.

As I read it, you could say something like:  If called on the original object, returns the obs coordinates for the entire dataset query. If called post-split, will return the subset of obs coordinates for the split. If called in a worker process, will return the obs coordinates for that individual worker.

Reply:
this works for me.